core/src/main/clojure/xtdb/sql.clj

```diff
@@ -11,12 +11,18 @@
            java.util.HashMap
            org.apache.arrow.memory.BufferAllocator))
 
+(defn parse-query
+  [query]
+  (binding [r/*memo* (HashMap.)]
```

Comment:
worth binding this to `(or r/*memo* (HashMap.))` so that you don't use a different memo for parse + analyse in the compile case?

(might have to explicitly def `r/*memo*` to `nil` so that it's not `Var$Unbound`, if it's not already?)